src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -140,14 +140,16 @@ sealed class NumericRange[T](
 
   override def take(n: Int): NumericRange[T] = {
     if (n <= 0 || isEmpty) newEmptyRange(start)
-    else if (n >= length) this
-    else new NumericRange.Inclusive(start, locationAfterN(n - 1), step)
+    else if (num.lt(locationAfterN(n - 1), end))
```

Comment:
This will overflow in case of `Int` or `Long`.

For instance, consider `NumericRange(2000000000, 2100000000, 70000000).take(20)`.